### DISCUSSION TITLE
Node-red does not request permission to Control Oven

### DISCUSSION THREAD

**Discussion by leifnel** [alexkn/node-red-contrib-homeconnect#35] [General]
When doing the "Start authorization" node-red does not request permission to Control Oven.
Only "Monitor your oven" and "Get and modify your oven settings"

So I can not set programs
![image](https://user-images.githubusercontent.com/9819287/110378472-ee171e00-8055-11eb-8f88-6d22ea52f1c6.png)

I bragged to wife I could program sequences. "10 minutes at 240C, then  40 minutes at 180C"
Wife is disappointed...

**Comment by leifnel** [comment]
Oh. I think I finally understand that my problem is "Additional Partner Agreement required" to control oven

It appears that's not possible for non-businesses. ?!?

**Reply by alexkn** [reply]
I don't have a device which needs that.
I think you should ask [developer support](https://developer.home-connect.com/support/contact)

Please share your experiences here.

**Reply by leifnel** [reply]
I got this from developer support:

> As I understand what you are planning to create is for personal use and not aimed at a wider audience, so partnership is not really a possibility in this case.
> 
> Due to considerable feedback regarding the want for the Oven-Control scope to become a feature available to all users of the API, recently we have made the decision to go ahead and provide this feature. I do not have a concrete timeline on when this will get implemented but all of our HC API users should receive Oven-Control by the end of the 1st half of 2021. Feel free to follow our newsletter, as this is definitely something we will announce once it goes live. You can toggle your newsletter subscription in your account settings on the Home Connect Developer portal.

So we'll just have to wait.
